src/screens/SignTransactionSheet.tsx

```diff
@@ -0,0 +1,2300 @@
+
+        {!isMessageRequest && (
+          <GasSpeedButton
+            marginTop={24}
+            horizontalPadding={20}
+            currentNetwork={currentNetwork}
+            theme={'dark'}
+            marginBottom={0}
+            asset={undefined}
+            fallbackColor={undefined}
+            testID={undefined}
+            showGasOptions={undefined}
+            validateGasParams={undefined}
+            crossChainServiceTime={undefined}
+          />
+        )}
+      </Box>
+    </Inset>
+  );
+};
+
+const SimulationCard = ({
+  simulation,
+  isPersonalSign,
+  currentNetwork,
+  simulationError,
+  isBalanceEnough,
+  walletBalance,
+}: {
+  simulation: TransactionSimulationResult | undefined;
+  isPersonalSign: boolean;
+  currentNetwork: Network;
+  simulationError: TransactionErrorType | undefined;
+  isBalanceEnough: boolean;
+  walletBalance: {
+    amount: string | number;
+    display: string;
+    symbol: string;
+  };
+}) => {
+  const cardHeight = useSharedValue(COLLAPSED_CARD_HEIGHT);
+  const contentHeight = useSharedValue(
+    COLLAPSED_CARD_HEIGHT - CARD_BORDER_WIDTH * 2
+  );
+  const spinnerRotation = useSharedValue(0);
+
+  const itemCount =
+    (simulation?.in?.length || 0) +
+    (simulation?.out?.length || 0) +
+    (simulation?.approvals?.length || 0);
+  const simulationUnavailable =
+    isPersonalSign || currentNetwork === Network.zora;
+
+  const noChanges = simulation && itemCount === 0;
+
+  const listStyle = useAnimatedStyle(() => ({
+    opacity: interpolate(
+      cardHeight.value,
+      [
+        COLLAPSED_CARD_HEIGHT,
+        contentHeight.value + CARD_BORDER_WIDTH * 2 > MAX_CARD_HEIGHT
+          ? MAX_CARD_HEIGHT
+          : contentHeight.value + CARD_BORDER_WIDTH * 2,
+      ],
+      [0, 1]
+    ),
+  }));
+
+  const spinnerStyle = useAnimatedStyle(() => {
+    return {
+      transform: [{ rotate: `${spinnerRotation.value}deg` }],
+    };
+  });
+
+  useEffect(() => {
+    if (!isBalanceEnough || simulationUnavailable || simulationError) return;
+    if (!isNil(simulation)) {
+      spinnerRotation.value = withTiming(360, timingConfig);
+    } else {
+      spinnerRotation.value = withRepeat(
+        withTiming(360, rotationConfig),
+        -1,
+        false
+      );
+    }
+  }, [
+    isBalanceEnough,
+    simulation,
+    simulationError,
+    simulationUnavailable,
+    spinnerRotation,
+  ]);
+
+  const renderSimulationEventRows = () => {
+    return (
+      <>
+        {simulation?.approvals?.map(change => {
+          return (
+            <SimulatedEventRow
+              key={`${change?.asset?.assetCode}-${change?.quantityAllowed}`}
+              amount={change?.quantityAllowed || '10'}
+              asset={change?.asset}
+              eventType="approve"
+            />
+          );
+        })}
+        {simulation?.out?.map(change => {
+          return (
+            <SimulatedEventRow
+              key={`${change?.asset?.assetCode}-${change?.quantity}`}
```

Comment:
yo I know this is merged already was curious how we implemented the same thing lol

and not that it matters a lot but I think this key is gonna conflict in a case like opensea pro buying lets say rainbow zorbs in bulk, same asset code and quantity, but from multiple sellers

idk how to do it either I added the index to the key even tho it's not good practice, considering this array won't change it's probably safe (unless we refetch and even then its hard to hit)